spec/Stratum.md

```diff
@@ -0,0 +1,137 @@
+Stratum is a light-weight mining protocol.
+
+# List of methods
+
+ * [mining.subscribe](#mining.subscribe)
+ * [mining.authorize](#mining.authorize)
+ * [mining.notify](#mining.notify)
+ * [mining.submit](#mining.submit)
+
+# Specification
+
+## mining.subscribe
+
+Used for subscribing mining jobs.
+
+Params: No parameters
+
+Return Type: No parameters
+
+Request Example
+```
+{
+    "jsonrpc": "2.0",
+    "id": 1,
+    "method": "mining.subscribe",
+    "params": []
+}
+```
+
+Response Example
+```
+{
+    "jsonrpc": "2.0",
+    "id": 1,
+    "result": [],
+    "error": null
+}
+```
+
+## mining.authorize
+
+Used for authorizing miner.
+
+Params:
+ 1. name: `string`
+ 2. password: `string`
+
+Return Type: `bool`
+
+Request Example
+```
+{
+    "jsonrpc": "2.0",
+    "id":2,
+    "method": "mining.authorize",
+    "params": ["miner1", "password"]
+}
+```
+
+Response Example
+```
+{
+    "jsonrpc": "2.0",
+    "id": 2,
+    "result": true,
+    "error": null
+}
+```
+
+## mining.notify
+
+Used for sending notifications with mining jobs
+
+Params: `WorkObject`
+
+Notification Example
+```
+{
+    "jsonrpc": "2.0",
+    "id": 3,
+    "method": "mining.notify",
+    "params": {
+        "0x56642f04d519ae3262c7ba6facf1c5b11450ebaeb7955337cfbc45420d573077",
+        "100"
+    },
+}
+```
+
+## mining.submit
+
+Used for submitting a proof-of-work solution.
+
+Params:
+ 1. powHash: `string`
+ 2. seal: `string[]`
+
+Return Type: `bool`
+
+Request Example
+```
+{
+    "jsonrpc": "2.0",
+    "id": 4,
+    "method": "mining.submit",
+    "params": ["0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef", ["0x56642f04d519ae3262c7ba6facf1c5b11450ebaeb7955337cfbc45420d573077"]],
+}
+```
+
+Response Example
+```
+{
+    "jsonrpc": "2.0",
+    "id": 4,
+    "result": true,
+    "error": null
+}
+```
+
+## Exception Handling (DRAFT)
+Stratum defines simple exception handling. Example of rejected share looks like:
```

Comment:
Example of rejected share looks like: -> Example of a rejected share looks like: